clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/PartitionLostCompleteEvent.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.consumer.internals.events;
+
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.KafkaException;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.SortedSet;
+
+public class PartitionLostCompleteEvent extends ApplicationEvent {
+
+    private final SortedSet<TopicPartition> lostPartitions;
+    private final Optional<KafkaException> error;
+
+    public PartitionLostCompleteEvent(SortedSet<TopicPartition> lostPartitions, Optional<KafkaException> error) {
+        super(Type.PARTITION_LOST_COMPLETE);
+        this.lostPartitions = Collections.unmodifiableSortedSet(lostPartitions);
+        this.error = error;
+    }
+
+    public SortedSet<TopicPartition> lostPartitions() {
+        return lostPartitions;
+    }
+
+    public Optional<KafkaException> error() {
+        return error;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        if (!super.equals(o)) return false;
+
+        PartitionLostCompleteEvent that = (PartitionLostCompleteEvent) o;
+
+        return lostPartitions.equals(that.lostPartitions) && error.equals(that.error);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = super.hashCode();
+        result = 31 * result + lostPartitions.hashCode();
+        result = 31 * result + error.hashCode();
+        return result;
+    }
+
+    @Override
+    protected String toStringBase() {
+        return super.toStringBase() + ", lostPartitions=" + lostPartitions + ", error=" + error;
+    }
+
+    @Override
+    public String toString() {
+        return getClass().getSimpleName() + "{" +
+                toStringBase() +
+                '}';
+    }
```

Comment:
This seems to be a quite complicated way to implement `toString`. When I read it, I cannot even tell how it will look like... Why don't we simply implement toString like we do in all classes in the code base?

Reply:
I think this is from Jun's comment in one of the commit that we should try to reuse the toString from its inherited class.